libs/@blockprotocol/graph/src/types/subgraph/edges/variants/knowledge.ts

```diff
@@ -71,19 +64,36 @@ export const isIncomingLinkEdge = (
   return outwardEdge.kind === "HAS_RIGHT_ENTITY" && outwardEdge.reversed;
 };
 
-export type ConstrainsPropertiesOnEdge = Subtype<
-  OntologyOutwardEdge,
+export type IsOfTypeEdge = Subtype<
+  GenericOutwardEdge,
   {
     reversed: false;
-    kind: "CONSTRAINS_PROPERTIES_ON";
+    kind: "IS_OF_TYPE";
     rightEndpoint: OntologyTypeVertexId;
   }
 >;
 
-export const isConstrainsPropertiesOnEdge = (
+export const isIsOfTypeEdge = (
   outwardEdge: OutwardEdge,
-): outwardEdge is ConstrainsPropertiesOnEdge => {
-  return (
-    outwardEdge.kind === "CONSTRAINS_PROPERTIES_ON" && !outwardEdge.reversed
-  );
+): outwardEdge is IsOfTypeEdge => {
+  return outwardEdge.kind === "IS_OF_TYPE" && !outwardEdge.reversed;
 };
+
+export type KnowledgeGraphOutwardEdge =
+  | OutgoingLinkEdge
+  | IncomingLinkEdge
+  | HasLeftEntityEdge
+  | HasRightEntityEdge
+  | IsOfTypeEdge;
```

Comment:
I did this because it turns out that expressing the things like below (`GenericOutwardEdge<KnowledgeGraphEdgeKind, boolean, EntityIdAndTimestamp>`) meant they just _would not_ play nicely with advanced types such as `Extract`.

Those types end up being pretty important in the ergonomics of using the `Subgraph` so I decided it'd be better to be fully explicit and stick to plain boring typescript here without generic stuff. 

This has had the side-effect benefit of having standardised names for the various edge permutations now, which we didn't have before.